sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/ObjectAggregationIterator.scala

```diff
@@ -111,25 +111,17 @@ class ObjectAggregationIterator(
     }
   }
 
+  private lazy val bufferFieldTypes =
+    aggregateFunctions.flatMap(_.aggBufferAttributes.map(_.dataType))
+
   // Creates a new aggregation buffer and initializes buffer values. This function should only be
-  // called under two cases:
-  //
-  //  - when creating aggregation buffer for a new group in the hash map, and
-  //  - when creating the re-used buffer for sort-based aggregation
+  // called when creating aggregation buffer for a new group in the hash map.
   private def createNewAggregationBuffer(): SpecificInternalRow = {
-    val bufferFieldTypes = aggregateFunctions.flatMap(_.aggBufferAttributes.map(_.dataType))
     val buffer = new SpecificInternalRow(bufferFieldTypes)
-    initAggregationBuffer(buffer)
```

Comment:
what's wrong with `initAggregationBuffer`?

Reply:
it's the duplicate code and the code has poor performance due to the scala sugar. `AggregationIterator.initializeBuffer` is effective and also used by `SortBasedAggregationIterator`